testsuite/cluster-test/src/cluster_swarm/cluster_swarm_kube.rs

```diff
@@ -54,6 +54,17 @@ impl ClusterSwarmKube {
         Ok(Self { client, node_map })
     }
 
+    fn service_spec(&self, peer_id: String) -> Result<Service> {
```

Comment:
So looks like this would only fail if the project itself contains incorrect yaml file?
In this case it probably will simplify interface if you just return Service and panic if yaml/json fails
I would say it's ok to simply panic if program itself is shipped with incorrect spec files